pkg/controller/controller_ref_manager.go

```diff
@@ -346,6 +346,7 @@ func (m *ReplicaSetControllerRefManager) AdoptReplicaSet(rs *extensions.ReplicaS
 	if err := m.canAdopt(); err != nil {
 		return fmt.Errorf("can't adopt ReplicaSet %v/%v (%v): %v", rs.Namespace, rs.Name, rs.UID, err)
 	}
+	rs.Finalizers = append(rs.Finalizers, metav1.FinalizerDeleteDependents)
```

Comment:
@caesarxuchao I thought we did this finalizer manipulation in the REST storage layer.